cirq-core/cirq/devices/superconducting_qubits_noise_properties.py

```diff
@@ -0,0 +1,269 @@
+# limitations under the License.
+
+from dataclasses import dataclass, field
+from typing import Dict, TYPE_CHECKING, List, Set
+
+from cirq import ops, protocols, devices
+from cirq.devices.noise_utils import (
+    OpIdentifier,
+    decoherence_pauli_error,
+)
+
+if TYPE_CHECKING:
+    import cirq
+
+SINGLE_QUBIT_GATES: Set[type] = {
+    ops.ZPowGate,
+    ops.PhasedXZGate,
+    ops.MeasurementGate,
+    ops.ResetChannel,
+}
+SYMMETRIC_TWO_QUBIT_GATES: Set[type] = {
+    ops.FSimGate,
+    ops.PhasedFSimGate,
+    ops.ISwapPowGate,
+    ops.CZPowGate,
+}
+ASYMMETRIC_TWO_QUBIT_GATES: Set[type] = set()
+TWO_QUBIT_GATES = SYMMETRIC_TWO_QUBIT_GATES | ASYMMETRIC_TWO_QUBIT_GATES
+
+
+# TODO: missing per-device defaults
+@dataclass
+class SuperconductingQubitsNoiseProperties(devices.NoiseProperties):
+    """Noise-defining properties for a quantum device.
+
+    Args:
+        gate_times_ns: Dict[type, float] of gate types to their duration on
+            quantum hardware.
+        t1_ns: Dict[cirq.Qid, float] of qubits to their T_1 time, in ns.
+        tphi_ns: Dict[cirq.Qid, float] of qubits to their T_phi time, in ns.
+        ro_fidelities: Dict[cirq.Qid, np.ndarray] of qubits to their readout
+            fidelity matrix.
+        gate_pauli_errors: dict of OpIdentifiers (a gate and the qubits it
+            targets) to the Pauli error for that operation. Keys in this dict
+            must have defined qubits.
+        validate: If True, performs validation on input arguments. Defaults
+            to True.
+    """
+
+    gate_times_ns: Dict[type, float]
+    t1_ns: Dict['cirq.Qid', float]
+    tphi_ns: Dict['cirq.Qid', float]
+    ro_fidelities: Dict['cirq.Qid', List[float]]
+    gate_pauli_errors: Dict[OpIdentifier, float]
+
+    validate: bool = True
+    _qubits: List['cirq.Qid'] = field(init=False, default_factory=list)
+    _depolarizing_error: Dict[OpIdentifier, float] = field(init=False, default_factory=dict)
+
+    def __post_init__(self):
+        if not self.validate:
+            return
+        t1_qubits = set(self.t1_ns)
+        tphi_qubits = set(self.tphi_ns)
+
+        if t1_qubits != tphi_qubits:
+            raise ValueError('Keys specified for T1 and Tphi are not identical.')
+
+        # validate two qubit gate errors.
+        self._validate_symmetric_errors('gate_pauli_errors')
+
+    def _validate_symmetric_errors(self, field_name: str) -> None:
+        gate_error_dict = getattr(self, field_name)
+        for op_id in gate_error_dict:
+            if len(op_id.qubits) != 2:
+                # single qubit op_ids also present, or generic values are
+                # specified. Skip these cases
+                if len(op_id.qubits) > 2:
+                    raise ValueError(
+                        f'Found gate {op_id.gate_type} with {len(op_id.qubits)} qubits. '
+                        'Symmetric errors can only apply to 2-qubit gates.'
+                    )
```

Comment:
That would break the `elif` and `else` blocks below. There might be a way to finagle it such that no nested ifs are required, but I suspect it would be less readable than this.